openedx/core/djangoapps/content/search/documents.py

```diff
@@ -576,12 +578,21 @@ def searchable_doc_for_container(
     }
 
     try:
-        container = lib_api.get_container(container_key)
+        container_obj = lib_api.get_container_from_key(container_key)
```

Comment:
The docstring for `get_container_from_key` says it's an "Internal method", but in #36504 we added it to `__all__` so it's now also part of the public API.

The thing is, it doesn't make sense as a public API since both `get_container` and `get_container_from_key` take a container key as their first argument. I think we should rename it to `_get_container_from_key`, and keep it as internal (remove it from `__all__`) and create a separate public function called `get_container_obj` which wraps `_get_container_from_key`. This would be much cleaner and clearer.

Reply:
I ended up being able to remove `get_container_from_key` from the public API without having to add a `get_container_obj` method. I had to add `container_pk` to `ContainerMetadata` (since ContainerLink needed it), but this seems cleaner to me too.